applications/nrf_desktop/src/modules/failsafe.c

```diff
@@ -43,9 +53,11 @@ static void failsafe_erase(void)
 	}
 }
 
-static void failsafe_clear(void)
+static int failsafe_clear(void)
 {
-	nrfx_reset_reason_clear(nrfx_reset_reason_get());
+	int err = hwinfo_clear_reset_cause();
```

Comment:
`return hwinfo_clear_reset_cause();` - for simplicity?

Reply:
I added some additional error logs